docs/installation.md

```diff
@@ -0,0 +1,94 @@
+# Installation and Local Development
+
+This guide explains how to **build Blaze4J from source** and run the official JSON Schema Test Suite for all supported drafts.
+
+---
+
+## Table of Contents
+
+- [Prerequisites](#prerequisites)
+- [1. Initialize Blaze Submodule](#1-initialize-blaze-submodule)
+- [2. Build Native Libraries](#2-build-native-libraries)
+  - [Windows](#windows)
+  - [Linux](#linux)
+  - [macOS](#macos)
+- [3. JSON Schema Test Suite (Drafts)](#3-json-schema-test-suite-drafts)
+  - [A. Add Test Suite Submodule](#a-add-test-suite-submodule)
+  - [B. Run Draft Test Runners](#b-run-draft-test-runners)
+- [Notes](#notes)
+
+---
+
+## Prerequisites
+
+Make sure you have the following tools installed:
+
+- **Java Development Kit (JDK) 11** or higher
+- **CMake 3.10** or higher
+- **C++ compiler** with C++11 support
+- **Git** (for submodules)
+
+---
+
+## 1. Initialize Blaze Submodule
+
+Clone the repository
+```
+git clone https://github.com/MadhavDhatrak/Blaze4J.git
+```
+
+ then run:
+```bash
+git submodule add https://github.com/sourcemeta/blaze.git deps/blaze
```

Comment:
I think this submodule setup needs to be improved. The very point of git submodules is that you shouldn't need to clone and add them all the time. Also a commit of a parent project clearly references a commit of the submodule (hence situations where a change in the submodule breaks the parent project are avoided).

In the end, the contributors should be able to obtain blaze4j and the respective blaze version in one command (details here: https://stackoverflow.com/questions/3796927/how-do-i-git-clone-a-repo-including-its-submodules )